internal/controller/lcore/assets.go

```diff
@@ -0,0 +1,1202 @@
+	metaLabels["app.kubernetes.io/component"] = "metrics"
+	metaLabels["openshift.io/user-monitoring"] = "true"
+
+	valFalse := false
+	serverName := strings.Join([]string{utils.OLSAppServerServiceName, r.GetNamespace(), "svc"}, ".")
+
+	serviceMonitor := monv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      utils.AppServerServiceMonitorName,
+			Namespace: r.GetNamespace(),
+			Labels:    metaLabels,
+		},
+		Spec: monv1.ServiceMonitorSpec{
+			Endpoints: []monv1.Endpoint{
+				{
+					Port:     "https",
+					Path:     utils.AppServerMetricsPath,
+					Interval: "30s",
+					Scheme:   "https",
+					TLSConfig: &monv1.TLSConfig{
+						CAFile:   "/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt",
+						CertFile: "/etc/prometheus/secrets/metrics-client-certs/tls.crt",
+						KeyFile:  "/etc/prometheus/secrets/metrics-client-certs/tls.key",
+						SafeTLSConfig: monv1.SafeTLSConfig{
+							InsecureSkipVerify: &valFalse,
+							ServerName:         &serverName,
+						},
+					},
+					Authorization: &monv1.SafeAuthorization{
+						Type: "Bearer",
+						Credentials: &corev1.SecretKeySelector{
+							Key: "token",
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: utils.MetricsReaderServiceAccountTokenSecretName,
+							},
+						},
+					},
+				},
+			},
+			JobLabel: "app.kubernetes.io/name",
+			Selector: metav1.LabelSelector{
+				MatchLabels: utils.GenerateAppServerSelectorLabels(),
+			},
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(cr, &serviceMonitor, r.GetScheme()); err != nil {
+		return nil, err
+	}
+
+	return &serviceMonitor, nil
+}
+
+func GeneratePrometheusRule(r reconciler.Reconciler, cr *olsv1alpha1.OLSConfig) (*monv1.PrometheusRule, error) {
+	metaLabels := utils.GenerateAppServerSelectorLabels()
+	metaLabels["app.kubernetes.io/component"] = "metrics"
+
+	rule := monv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      utils.AppServerPrometheusRuleName,
+			Namespace: r.GetNamespace(),
+			Labels:    metaLabels,
+		},
+		Spec: monv1.PrometheusRuleSpec{
+			Groups: []monv1.RuleGroup{
+				{
+					Name: "ols.operations.rules",
+					Rules: []monv1.Rule{
+						{
+							Record: "ols:rest_api_query_calls_total:2xx",
+							Expr:   intstr.FromString("sum by(status_code) (ols_rest_api_calls_total{path=\"/v1/streaming_query\",status_code=~\"2..\"})"),
+							Labels: map[string]string{"status_code": "2xx"},
+						},
+						{
+							Record: "ols:rest_api_query_calls_total:4xx",
+							Expr:   intstr.FromString("sum by(status_code) (ols_rest_api_calls_total{path=\"/v1/streaming_query\",status_code=~\"4..\"})"),
+							Labels: map[string]string{"status_code": "4xx"},
+						},
+						{
+							Record: "ols:rest_api_query_calls_total:5xx",
+							Expr:   intstr.FromString("sum by(status_code) (ols_rest_api_calls_total{path=\"/v1/streaming_query\",status_code=~\"5..\"})"),
+							Labels: map[string]string{"status_code": "5xx"},
```

Comment:
s/ols_/ls_/

Reply:
again, this is identical to current implementation